### DISCUSSION TITLE
KeyError: "Node '1' does not have the requested weight field."

### DISCUSSION THREAD

**Discussion by vappiah** [networkx/networkx#5428] [Q&A] [answered]
Hello Networkx Developer,
I have a undirected weight graph and I want to get the clique with the weight. I tried using the _max_weight_clique_ function but I got the error message: _KeyError: "Node '1' does not have the requested weight field."_

I am using networkx 2.7.1

Minimal reproducing example:

```python
G = nx.Graph()
G.add_nodes_from(["1", "2", "3", "4", "5"])
weighted_edges = [
    ("1", "2", 0.11),
    ("1", "3", 3.1),
    ("1", "5", 2.25),
    ("4", "5", 0.25),
    ("2", "5", 0.2),
    ("2", "4", 0.22),
    ("2", "3", 0.2),
]
G.add_weighted_edges_from(weighted_edges)
max_weight_clique = nx.max_weight_clique(G)
```

Please advice.
Thanks

**Comment by rossbar** [comment, marked as answer]
Hi @vappiah 

First, a quick suggestion re: how best to ask for help: instead of linking to gists, it is preferable to provide your example code directly in the issue/discussion. You can do this by wrapping your example in a "fence" of triple backticks:

````
```python
<your example code goes here>
```
````
The `python` is included at the start of the fence so that the syntax highlighting is correct.

It's also helpful to have the example code be as *minimal* as possible - that way, it's much easier to focus on the lines that are actually causing the problem.

I've gone ahead and modified your original comment to account for these suggestions; but keep this in mind when asking future questions. And thanks for the nice question!

Now, on to the actual question:

The problem here is that `max_weight_clique` expects the *nodes* to be weighted, but you've only provided *edge* weights. A careful reading of the [docstring](https://networkx.org/documentation/latest/reference/algorithms/generated/networkx.algorithms.clique.max_weight_clique.html) should help clairify:

> The *weight* of a clique is the sum of the weights of its nodes.

**Reply by vappiah** [reply]
@rossbar Thanks for the response
I am looking for the maximum weighted edge clique.  So what I will do is to use find_cliques function to find all cliques and compute the weights.
